src/proof_system/preprocess.rs

```diff
@@ -222,6 +224,7 @@ impl<E: PairingEngine, P: TEModelParameters<BaseField = E::Fr>>
     /// must perform IFFTs on the selector polynomials and permutation
     /// polynomials in order to commit to them and have the same transcript
     /// view.
+    #[allow(clippy::type_complexity)] // FIXME: Add struct for preprocessed key.
```

Comment:
ditto.

Reply:
Note how this lint indicates a change to be made rather than an oversight on the linter. Those two kinds of situations are important to distinguish between. See my general comments above about linting.

Reply:
Ditto response above.